common/config/azure-pipelines/templates/run-eslint-deprecation.yaml

```diff
@@ -0,0 +1,64 @@
+steps:
+  - bash: |
+      newPackageVersion=$(cat common/config/rush/version-policies.json | grep \"version\": | awk -F'"' '{split($4,a,"-"); print a[1]}')
+      sed -i "s/\(addVersion: \"\)[^\"]*/\1$newPackageVersion/" common/config/eslint/eslint.config.deprecation-policy.js
+      echo "##vso[task.setvariable variable=version;isOutput=true]$newPackageVersion"
+    displayName: Change addVersion in ESLint config to current version
+    name: getNewVersion
+
+  - bash: node common/scripts/install-run-rush.js lint-deprecation
+    displayName: rush lint-deprecation
+
+  - bash: git add .
+    displayName: git add .
+
+  - bash: |
+      git commit -m "Apply deprecation date rule for v$(getNewVersion.version)"
```

Comment:
Have you tested this with no new deprecations (where the rule didn't make any changes)? Won't the pipeline error out if the staging area is empty?

Reply:
i thought this was his test 2 situation. If not then I misunderstood